src/components/Data/Collections/Watch.vue

```diff
@@ -49,51 +129,275 @@
 
 <script>
   import Headline from '../../Materialize/Headline'
+  import JsonFormatter from '../../../directives/json-formatter.directive'
+  import ScrollGlue from '../../../directives/scroll-glue.directive'
   import jQueryCollapsible from '../../Materialize/collapsible'
-  import { subscribe, unsubscribe, clear } from '../../../vuex/modules/data/actions'
-  import { notifications } from '../../../vuex/modules/data/getters'
   import Notification from '../Realtime/Notification'
+  import SubscriptionControls from '../Realtime/SubscriptionControls'
   import CollectionDropdown from '../Collections/Dropdown'
+  import Filters from '../../Common/Filters/Filters'
+  import kuzzle from '../../../services/kuzzle'
+  import { setBasicFilter } from '../../../vuex/modules/list/actions'
+  import { rawFilter, basicFilter, basicFilterForm } from '../../../vuex/modules/list/getters'
+  import { availableFilters, formatFromBasicSearch } from '../../../services/filterFormatRealtime'
 
   export default {
     name: 'CollectionWatch',
     props: {
       index: String,
       collection: String
     },
+    vuex: {
+      actions: {
+        setBasicFilter
+      },
+      getters: {
+        rawFilter,
+        basicFilter,
+        basicFilterForm
+      }
+    },
     data () {
       return {
         subscribed: false,
-        room: null
+        room: null,
+        filters: {},
+        availableFilters,
+        formatFromBasicSearch,
+        subscribeOptions: {scope: 'all', users: 'all', state: 'all'},
+        notifications: [],
+        notificationsLengthLimit: 50,
+        warning: {message: '', count: 0, lastTime: null, info: false},
+        scrollGlueActive: true,
+        scrollListener: null
+      }
+    },
+    watch: {
+      index () {
+        this.reset()
+      },
+      collection () {
+        this.reset()
+      }
+    },
+    ready () {
+      // display the toolbar when user scroll (or when scroll glue is active)
+      let scrolled = false
+      let toolbar = document.getElementById('notification-controls-fixed')
+
+      this.notifications = []
+
+      window.onscroll = function () {
+        scrolled = true
+      }
+
+      // delay position checking in an interval instead of onscroll event to reduce lags
+      this.scrollListener = setInterval(function () {
```

Comment:
This is the logic for scrollglue directive. We don't have to put this logic in parent component, because if several components need this directive, we have to duplicate code.